src/main/java/ch/heigvd/pro/b04/questions/QuestionController.java

```diff
@@ -100,6 +97,71 @@ public QuestionController(QuestionRepository repository,
         .collect(Collectors.toList());
   }
 
+  /**
+   * gets a {@link ServerQuestion} by his id.
+   *
+   * @param token       token of the moderator or participant
+   * @param idModerator id of moderator owning the poll
+   * @param idPoll      part of the {@link ServerPollIdentifier}
+   * @param idQuestion  part of the {@link ServerQuestionIdentifier}
+   * @return {@link ServerQuestion} found
+   * @throws ResourceNotFoundException if one of the parameters is broken
+   * @throws WrongCredentialsException if there a credentials violation
+   */
+  @GetMapping(value = "/mod/{idModerator}/poll/{idPoll}/question/{idQuestion}")
+  @Transactional
+  public ServerQuestion byId(
+      @RequestParam(name = "token") String token,
+      @PathVariable(name = "idModerator") int idModerator,
+      @PathVariable(name = "idPoll") int idPoll,
+      @PathVariable(name = "idQuestion") int idQuestion)
+      throws ResourceNotFoundException, WrongCredentialsException, PollNotExistingException {
+    ServerQuestionIdentifier idToFind = ServerQuestionIdentifier.builder()
+        .idServerQuestion(idQuestion)
+        .idxPoll(pollRepository.findById(ServerPollIdentifier.builder().idPoll(idPoll).idxModerator(
+            moderatorRepository.findById(idModerator).orElseThrow(ResourceNotFoundException::new))
+            .build())
+            .orElseThrow(ResourceNotFoundException::new))
+        .build();
+
+    ServerQuestion questionWanted = repository.findById(idToFind)
+        .orElseThrow(ResourceNotFoundException::new);
+
+    for (ServerQuestion q : verifyModeratorOrParticipantAccess(participantRepository,
+        moderatorRepository, idPoll, token).getPollServerQuestions()) {
+      if (q.equals(questionWanted)) {
+        return q;
+      }
+    }
+    throw new ResourceNotFoundException();
+  }
+
+  private ServerPoll verifyModeratorOrParticipantAccess(ParticipantRepository prpRepo,
+      ModeratorRepository modoRepo, int idPoll, String token)
+      throws ResourceNotFoundException, PollNotExistingException {
+    ServerPoll pollTest;
+    Optional<Participant> pollT = prpRepo.findByToken(token);
+    //if token doesn't lead to a Participant...
+    if (pollT.isEmpty()) {
+      //...it looks for a moderator
+      Optional<Moderator> pollM = modoRepo.findByToken(token);
+      if (pollM.isEmpty()) {
+        throw new ResourceNotFoundException();
+      } else {
+        //poll inside pollSet of the modo
+        pollTest = pollM.get().searchPoll(ServerPollIdentifier.builder()
+            .idPoll(idPoll).idxModerator(pollM.get()).build());
+      }
```

Comment:
Just so you know, you can directly have a poll like this:
```
Moderator pollM = modRepo.findByToken(token).orElseThrow(ResourceNotFoundException:new);
```

Reply:
Yeah true